fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/ArmorRenderer.java

```diff
@@ -0,0 +1,105 @@
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * Armor renderers render worn armor items with custom code.
+ * They may be used to render armor with special models or effects.
+ *
+ * <p>The renderers are registered with {@link ArmorRenderingRegistry#register(ArmorRenderer, Item...)}.
+ */
+@FunctionalInterface
+@Environment(EnvType.CLIENT)
+public interface ArmorRenderer {
+	static final HashMap<Item, ArmorRenderer> RENDERERS = new HashMap<>();
+	/**
+	 * Renders an armor part
+	 *
+	 * @param matrices			the matrix stack
+	 * @param vertexConsumers	the vertex consumer provider
+	 * @param stack				the item stack of the armor item
+	 * @param entity			the entity wearing the armor item
+	 * @param slot				the equipment slot in which the armor stack is worn
+	 * @param light				packed lightmap coordinates
+	 * @param contextModel		the model provided by {@link FeatureRenderer#getContextModel()}
+	 */
+	void render(MatrixStack matrices, VertexConsumerProvider vertexConsumers, ItemStack stack, LivingEntity entity, EquipmentSlot slot, int light, BipedEntityModel<LivingEntity> contextModel);
+
+	/**
+	 * Helper method for rendering a specific armor model, comes after setting visibility.
+	 * <p>This primarily handles applying glint and the correct {@link RenderLayer}
+	 * @param matrices			the matrix stack
+	 * @param vertexConsumers	the vertex consumer provider
+	 * @param light				packed lightmap coordinates
+	 * @param stack				the item stack of the armor item
+	 * @param model				the model to be rendered
+	 * @param texture			the texture to be applied
+	 */
+	static void renderPart(MatrixStack matrices, VertexConsumerProvider vertexConsumers, int light, ItemStack stack, Model model, Identifier texture) {
+		VertexConsumer vertexConsumer = ItemRenderer.getArmorGlintConsumer(vertexConsumers, RenderLayer.getArmorCutoutNoCull(texture), false, stack.hasGlint());
+		model.render(matrices, vertexConsumer, light, OverlayTexture.DEFAULT_UV, 1.0F, 1.0F, 1.0F, 1.0F);
+	}
+
+	/**
+	 * Registers the armor renderer for the specified items
+	 * @param renderer	the renderer
+	 * @param items		the items
+	 * @throws IllegalArgumentException if an item already has a registered armor renderer
+	 * @throws NullPointerException if either an item or the renderer is null
+	 */
+	static void register(ArmorRenderer renderer, Item... items) {
+		for (Item item : items) {
+			Objects.requireNonNull(item, "armor item is null");
+			Objects.requireNonNull(renderer, "renderer is null");
+			if (RENDERERS.putIfAbsent(item, renderer) != null) {
+				throw new IllegalArgumentException("Custom armor renderer already exists for " + Registry.ITEM.getId(item));
+			}
+		}
+	}
+
+	/**
+	 * Standard getter method for armor renderers
+	 * @param item	the item, associated with the renderer
+	 * @return the renderer, returns null if no renderer is registered
+	 */
+	@Nullable
+	static ArmorRenderer getRenderer(Item item){
+		return RENDERERS.get(item);
+	}
```

Comment:
If the map is moved to an impl class, I'm not sure this will still be needed (this function can be moved to the impl class, and called directly by the mixin).